src/hats/pixel_math/healpix_shim.py

```diff
@@ -1,41 +1,85 @@
 from __future__ import annotations
 
+import math
+
 import astropy.units as u
+import cdshealpix
 import healpy as hp
 import numpy as np
-from astropy.coordinates import SkyCoord
+from astropy.coordinates import Latitude, Longitude, SkyCoord
 
 # pylint: disable=missing-function-docstring
 
 ## Arithmetic conversions
 
 
-def npix2order(param):
-    return hp.nside2order(hp.npix2nside(param))
+MAX_HEALPIX_ORDER = 29
+
+
+def is_order_valid(order: int) -> bool:
+    return np.all(0 <= order) and np.all(order <= MAX_HEALPIX_ORDER)
+
+
+def npix2order(npix: int) -> int:
+    if npix <= 0:
+        raise ValueError("Invalid value for npix")
+    order = int(math.log2(npix / 12)) >> 1
+    if not is_order_valid(order) or not 12 * (1 << (2 * order)) == npix:
+        raise ValueError("Invalid value for npix")
+    return order
+
+
+def order2nside(order: int) -> int:
+    if not is_order_valid(order):
+        raise ValueError("Invalid value for order")
+    return 1 << order
+
+
+def order2npix(order: int) -> int:
+    if not is_order_valid(order):
+        raise ValueError("Invalid value for order")
+    return 12 * (1 << (2 * order))
+
 
+def nside2resol(nside: int, arcmin: bool = False) -> float:
+    resol_rad = np.sqrt(nside2pixarea(nside))
 
-def order2nside(param):
-    return hp.order2nside(param)
+    if arcmin:
+        return np.rad2deg(resol_rad) * 60
 
+    return resol_rad
 
-def order2npix(param):
-    return hp.order2npix(param)
 
+def nside2pixarea(nside: int, degrees: bool = False) -> float:
+    npix = 12 * nside * nside
+    pix_area_rad = 4 * np.pi / npix
+    if degrees:
+        return pix_area_rad * (180 / np.pi) * (180 / np.pi)
+    return pix_area_rad
 
-def nside2resol(*args, **kwargs):
-    return hp.nside2resol(*args, **kwargs)
 
+def ang2pix(nside: int, theta: float, phi: float, nest: bool = False, lonlat: bool = False) -> int:
```

Comment:
If `nest=False` is not supported, can we make `nest=True` the default? 

And we typically also set `lonlat=True`. 

And it would be friendlier to use `order` as an input, as we don't do much with nside.

So maybe what I'm seeing is really a different method...

`def radec2pixel(ra, dec, order)`

Reply:
Sounds good! I was going for as close to healpy as possible, but this sounds better